File: src/server/InvestmentApp-Server/Models/IndustryCriteria.cs
using System;
using System.ComponentModel.DataAnnotations.Schema;

namespace InvestmentApp.Models;

public class IndustryCriteria
{
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public Guid Id { get; set; }

    public Industry Industry { get; set; }

    public Guid IndustryId { get; set; }

    public Criteria Criteria { get; set; }

    public Guid CriteriaId { get; set; }

    public double IndustrySpecificWeight { get; set; }
}

File: src/client/src/app/models/industry-criteria.ts
import { DataTableRow } from ".";
import { Guid } from "guid-typescript";
import { Criteria } from "./criteria";
import { Industry } from "./industry";

export interface IndustryCriteria {
    id: Guid
    industry: Industry;
    industryId: Guid;
    criteria: Criteria;
    criteriaId: Guid;
    industrySpecificWeight: number;
}

export interface IndustryCriteriaRow extends DataTableRow {
    id: string,
    internalId: string;
    industry: string;
    industryId: string;
    criteria: string;
    criteriaId: string;
    industrySpecificWeight: number;
}
